modules/querying/pages/func/query-user-defined-functions.adoc

```diff
@@ -51,36 +49,104 @@ Below is a list of data types and where each is allowed in a function.
 | All other C{pp} data types | *No* | *No* | Yes
 |===
 
-If the function requires a user-defined struct or helper function, define it in the `ExprUtil` file you downloaded in Step 1.
+You can write your functions in the `ExprFunctions` file provided as a sample in TigerGraph Server installations, or create your own `.hpp` files from scratch.
+
+If your function requires a user-defined struct or helper function, that struct or helper function must be defined in a separate `ExprUtil` file.
 
-Below is an example of an `.hpp` file that contains two separate Query UDFs. The first function returns `true` if the passed value is greater than 3, and the second reverses a string.
+Below is an example of a short `ExprFunctions` file containing a single UDF that reverses a string. Note the `include` statement on the first line.
 
 .New code in ExprFunctions.hpp
 
 [source,c++]
 ----
 #include <algorithm>  // for std::reverse
-inline bool greater_than_three (double x) {
-  return x > 3;
-}
-
 inline string reverse(string str){
   std::reverse(str.begin(), str.end());
   return str;
 }
 ----
 
+== Use GitHub to store UDFs
+
+You can configure GSQL to read from a GitHub repository for ExprFunctions and ExprUtil.
+
+If GitHub access is configured, GSQL will retrieve user source code files from GitHub before files uploaded via `PUT`, so long as the files exist. 
```

Comment:
@a-m-thomas Is that the files need to exist or is it that only the configuration needs to be valid?